controllers/substitute.js

```diff
@@ -262,6 +272,10 @@ function saveToDB(passon) {
     });
  }
 
+ /**
+  * function for cleanup after error is detected
+  * @param {object} passon - compendium id and data of compendia
+  */
  function cleanup(passon, index) {
```

Comment:
What does `index` do here? I have not found a usage of "noCleanup" in the file, so if this is a not-yet used feature, I suggest not to implement it.